File: src/main/java/br/com/blz/api/models/ProductEntity.java
package br.com.blz.api.models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.validation.constraints.NotNull;



@Entity
@Table(name = "product")
public class ProductEntity implements Serializable {

	
	private static final long serialVersionUID = 1L;

	@Id
	@Column(name = "sku",length = 50, unique = true, nullable = false)
	private long sku;
	
	@NotNull
	private String name;
	
	@OneToOne(cascade = CascadeType.ALL)
	private InventoryEntity inventory;
	
	@Transient
	private boolean isMarketable;

	public long getSku() {
		return sku;
	}

	public void setSku(long sku) {
		this.sku = sku;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public InventoryEntity getInventory() {
		return inventory;
	}

	public void setInventory(InventoryEntity inventory) {
		this.inventory = inventory;
	}

	public boolean isMarketable() {
		return this.getInventory().getQuantity() > 0 ? true : false;
	}


}

File: src/main/kotlin/com/boticario/testkotlin/infrastructure/persistence/entity/ProductEntity.kt
package com.boticario.testkotlin.infrastructure.persistence.entity

import com.boticario.testkotlin.domain.entity.Inventory
import com.boticario.testkotlin.domain.entity.Product
import jakarta.persistence.CascadeType
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

@Entity
class ProductEntity(
        @Id
        private val sku: Long = 0L,
        private val name: String = "",

        @OneToMany(cascade = [CascadeType.ALL])
        private val warehouses: MutableList<WarehouseEntity> = mutableListOf()
) {
        fun toDomain() = Product(sku, name, Inventory(warehouses.map { it.toDomain() }.toMutableList()))


        companion object {
                fun fromDomain(product: Product) = ProductEntity(
                        product.sku,
                        product.name,
                        product.inventory.warehouses.map { WarehouseEntity.fromDomain(it) }.toMutableList()
                )
        }
}